common/djangoapps/entitlements/models.py

```diff
@@ -24,3 +24,25 @@ class CourseEntitlement(TimeStampedModel):
         help_text='The current Course enrollment for this entitlement. If NULL the Learner has not enrolled.'
     )
     order_number = models.CharField(max_length=128, null=True)
+
+    @classmethod
+    def get_active_user_course_entitlements(cls, user, course_uuid):
+        """
+        Returns all the available sessions for a given course.
+        """
+
+        try:
+            entitlement = cls.objects.get(
+                user=user,
+                course_uuid=course_uuid,
+            )
+            return entitlement
+        except cls.DoesNotExist:
+            return None
+
+    @classmethod
+    def set_enrollment(cls, entitlement, enrollment):
+        """
+        Fulfills an entitlement by specifying a session.
+        """
+        cls.objects.filter(id=entitlement.id).update(enrollment_course_run=enrollment)
```

Comment:
Should we do any validation here to make sure the entitlement hasn't already been expired?

Reply:
I think that check belongs at the higher levels that would be making this call.  Potentially before the enroll happens.  We will add this with integration of the policy management code.

Reply:
I am adding AC to the integration task to make sure we add in checks on enroll and any we think we need on switch session.

Reply:
Thinking about this some more.  I will let you know when I push up changes.